### PR TITLE
DOC-4550: differences between REGEXP_LIKE and REGEXP_CONTAINS

### PR DESCRIPTION
**Pull request by simon-dew**
Please see draft documentation here:

* [Pattern-matching Functions](https://simon-dew.github.io/docs-site/DOC-4550/server/6.0/n1ql/n1ql-language-reference/patternmatchingfun.html)

### CODE REVIEW

**Comment by simon-dew** [comment]
Clarified difference between REGEXP_LIKE and REGEXP_CONTAINS. Also made the examples for these two functions almost identical, so the difference between the results was clear.

File: modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc [hunk 1]
```diff
@@ -2,109 +2,129 @@
...
 
 [#section_regex_contains]
-== REGEXP_CONTAINS(expression, pattern)
+== REGEXP_CONTAINS(`expression`, `pattern`)
 
-*Arguments*::
-*expression*: string, or any N1QL expression that evaluates to a string
-+
-*pattern*: string representing a supported regular expression
+=== Arguments
```
Comment (jamesnocentini) [inline modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc:16]:
```suggestion
=== Arguments

```

File: modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc [hunk 2]
```diff
@@ -2,109 +2,129 @@
...
+expression:: String, or any N1QL expression that evaluates to a string.
 
-*Return Value*:: Returns True if the string value contains the regular expression pattern.
+pattern:: String representing a supported regular expression.
 
-*Example*::
-+
+=== Return Value
```
Comment (jamesnocentini) [inline modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc:22]:
```suggestion
=== Return Value

```

File: modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc [hunk 3]
```diff
@@ -2,109 +2,129 @@
...
+Returns TRUE if the string value contains any sequence that matches the regular expression pattern.
+
+=== Example
```
Comment (jamesnocentini) [inline modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc:26]:
```suggestion
=== Example

```

File: modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc [hunk 4]
```diff
@@ -2,109 +2,129 @@
...
   }
 ]
 ----
+====
 
 [#section_regex_like]
-== REGEXP_LIKE(expression, pattern)
+== REGEXP_LIKE(`expression`, `pattern`)
+
+=== Arguments
```
Comment (jamesnocentini) [inline modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc:63]:
```suggestion
=== Arguments

```

File: modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc [hunk 5]
```diff
@@ -2,109 +2,129 @@
...
+expression:: String, or any N1QL expression that evaluates to a string.
 
-*Arguments*::
-*expression*: string, or any N1QL expression that evaluates to a string
-+
-*pattern*: string representing a supported regular expression
+pattern:: String representing a supported regular expression.
 
-*Return Value*:: Returns True if the string value contains the regular expression pattern.
+=== Return Value
```
Comment (jamesnocentini) [inline modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc:69]:
```suggestion
=== Return Value

```

File: modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc [hunk 6]
```diff
@@ -2,109 +2,129 @@
...
+Returns TRUE if the string value exactly matches the regular expression pattern.
 
-*Example*::
-+
+=== Example
```
Comment (jamesnocentini) [inline modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc:73]:
```suggestion
=== Example

```

File: modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc [hunk 7]
```diff
@@ -2,109 +2,129 @@
...
   }
 ]
 ----
+====
 
 [#section_regex_position]
-== REGEXP_POSITION(expression, pattern)
+== REGEXP_POSITION(`expression`, `pattern`)
+
+=== Arguments
```
Comment (jamesnocentini) [inline modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc:110]:
```suggestion
=== Arguments

```

File: modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc [hunk 8]
```diff
@@ -2,109 +2,129 @@
...
+expression:: String, or any N1QL expression that evaluates to a string.
 
-*Arguments*::
-*expression*: string, or any N1QL expression that evaluates to a string
-+
-*pattern*: string representing a supported regular expression
+pattern:: String representing a supported regular expression.
 
-*Return Value*::
+=== Return Value
```
Comment (jamesnocentini) [inline modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc:116]:
```suggestion
=== Return Value

```

File: modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc [hunk 9]
```diff
@@ -2,109 +2,129 @@
...
 Returns first position of the occurrence of the regular expression _pattern_ within the input string _expression_.
 Returns -1 if no match is found.
 Position counting starts from zero.
 
-*Example*::
+=== Example
```
Comment (jamesnocentini) [inline modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc:122]:
```suggestion
=== Example

```

File: modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc [hunk 10]
```diff
@@ -125,34 +145,38 @@ LIMIT 2;
...
-*pattern*: string representing a supported regular expression
-+
-*repl*: string, or any N1QL expression that evaluates to a string
-+
-*n*: the maximum number of times to find and replace the matching pattern
-
-*Return Value*::
+== REGEXP_REPLACE(`expression`, `pattern`, `repl` [, `n`])
+
+=== Arguments
```
Comment (jamesnocentini) [inline modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc:162]:
```suggestion
=== Arguments

```

File: modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc [hunk 11]
```diff
@@ -125,34 +145,38 @@ LIMIT 2;
...
+expression:: String, or any N1QL expression that evaluates to a string.
+
+pattern:: String representing a supported regular expression.
+
+repl:: String, or any N1QL expression that evaluates to a string.
+
+n:: [Optional] The maximum number of times to find and replace the matching pattern.
+
+=== Return Value
```
Comment (jamesnocentini) [inline modules/n1ql/pages/n1ql-language-reference/patternmatchingfun.adoc:172]:
```suggestion
=== Return Value

```

**Review by jamesnocentini — CHANGES_REQUESTED** [review]
I've added suggestions to add blank lines after headings, even if it takes a bit more space. Does it make a difference to the output?

I got the same feedback from Dan https://github.com/couchbase/docs-ui/pull/49/files/10377566a2a76b0e81b48b18ecc92f8a9f53aaf0#diff-4f7b090e75307f1943e69b9bdfa0f534

**Review by jamesnocentini — APPROVED** [review]
Approving. Previous feedback is definitely not a blocker for this PR.

**Pull request merged**